test/rpc/conftest.py

```diff
@@ -2,11 +2,48 @@
 import pytest
 import random
 import time
-from typing import Dict, Any
+from typing import Dict, Any, Set
 
 import yaml
 
 
+def pytest_addoption(parser):
+    parser.addoption(
+        '--profile', default='postgres', help='Use the postgres profile',
+    )
+
+
+def _get_item_profiles(item) -> Set[str]:
+    supported = set()
+    for mark in item.iter_markers(name='supported'):
+        supported.update(mark.args)
+    return supported
+
+
+def pytest_collection_modifyitems(config, items):
+    selected_profile = config.getoption('profile')
+
+    to_remove = []
+
+    for item in items:
+        item_profiles = _get_item_profiles(item)
+        if selected_profile not in item_profiles and 'any' not in item_profiles:
+            to_remove.append(item)
+
+    for item in to_remove:
+        items.remove(item)
+
+
+def pytest_configure(config):
+    config.addinivalue_line('markers', 'supported(: Marks postgres-only tests')
```

Comment:
this is tiny, but do you mean `supported('postgres')` here?

Reply:
This is actually worse than that, I'll resolve it though - I think it should be `config.addinivalue_line('markers', 'supported(plugin, ...): Marks supported test types')` - Those other marks were sitting around from when I was trying another marking method.